### ISSUE TITLE
"Segmentation fault (core dumped)" error when performing multi-sample 2-pass mapping

### ISSUE THREAD

**Issue by brunogrande** [alexdobin/STAR#47]
I'm trying to perform 2-pass mapping on some canine RNA-seq libraries (STAR v2.4.1d). When I tested with one sample, everything worked fine. However, now that I'm extending to the entire cohort (29 samples), I'm running into "Segmentation fault (core dumped)" errors during the second pass. Here's an example command:

```
/extscratch/morinlab/projects/canine_lymphoma/software/star-2.4.1d/bin/Linux_x86_64_static/STAR \
	--readFilesCommand zcat \
	--genomeDir /extscratch/morinlab/reference/igenomes/Canis_familiaris/UCSC/canFam3/Sequence/STARIndex-2.4.1d/ \
	--outFileNamePrefix star_second_pass/ccb030084-0100_ \
	--sjdbFileChrStartEnd /extscratch/morinlab/projects/canine_lymphoma/results/star_pipeline-1.0.0/morin_cohort/2-concatenate_splice_junctions/all_samples_SJ.out.sort.tab \
	--outSAMtype BAM SortedByCoordinate \
	--runThreadN 5 \
	--sjdbOverhang 75 \
	--readFilesIn /extscratch/morinlab/projects/canine_lymphoma/data/fastq_rnaseq/morin_cohort/ccb030084-0100_R1.fastq.gz /extscratch/morinlab/projects/canine_lymphoma/data/fastq_rnaseq/morin_cohort/ccb030084-0100_R2.fastq.gz
```

At first, I thought it was caused by the presence of more than one SJ.out.tab files in the command, despite the documentation saying that this is supported. When I re-ran the command while only passing the SJ.out.tab file from one sample, it worked. 

Strangely though, when I concatenated the SJ.out.tab files from all samples (using the command below to ensure no duplicates), it started segfaulting again. I even ensured that the order of the splice junctions matched the order of the chromosomes in the reference genome (using a separate command), but that didn't help. 

```
cat ../2015-06-09_05-23-27/*_star_pipeline/outputs/star_first_pass/*_SJ.out.tab | cut -f1-4 | sort -k1,1 -k2,2n -k3,3n | uniq
```

Do you have any ideas as to why passing a SJ.out.tab file from one sample works, but passing a concatenated SJ.out.tab file from multiple samples ends up failing? Thanks!

**Comment by brunogrande** [comment]
I don't know if this issue is related to #46.

**Comment by brunogrande** [comment]
Based on [this thread](https://groups.google.com/forum/#!topic/rna-star/plZ54eEd12g) on the rna-star Google Group, I avoided the "on-the-fly" approach. Instead, I created a new genome index using the gene annotations (GTF file) and new splice junctions detected from the first pass (concatenated from all samples). I'm not seeing any segmentation faults. This is a workaround, but it would be nice if the on-the-fly method would work eventually, as it is convenient to avoid the genome reindexing as a separate step.
